### PR TITLE
Updates to migration tool

### PR DESCRIPTION
**Pull request by mickster04**
I made a small change to the setup instructions to make it clear they are to be followed in order
I also added more details to the dlrs migration tool including adding debug output of rules that couldn't be migrated, aligned to the flow field names that they should apply to in order to make it easier to migrate remaining rules to flow actions.

PLEASE NOTE, i couldn't get prettier to work so the formatting is wrong

### CODE REVIEW

File: scripts/convert-dlrs-rules.apex [hunk 1]
```diff
@@ -34,6 +30,37 @@ for (dlrs__LookupRollupSummary2__mdt dlrsRule : dlrs__LookupRollupSummary2__mdt.
...
+    unmigratableRule.put('Rollup Context','INSERT / UPDATE / UPSERT / DELETE: see documentation');
+    unmigratableRule.put('Lookup Field On Rollup Object','Id');
+    unmigratableRule.put('Lookup Field On Calc Item',dlrsRule.dlrs__RelationshipField__c);
+    unmigratableRule.put('Calc Item Rollup Field',dlrsRule.dlrs__FieldToAggregate__c);
+    unmigratableRule.put('Object for \"Prior records to rollup\"',dlrsRule.dlrs__ChildObject__c);
+    unmigratableRule.put('Action label',(String) dlrsRule.get('MasterLabel'));
+    unmigrateableRules.add(unmigratableRule);
+    continue;
+   }
+  Metadata.CustomMetadata customMetadata = new Metadata.CustomMetadata();
```
Comment (jamessimone) [inline scripts/convert-dlrs-rules.apex:59]: these lines shouldn't be part of the changes - they should remain where they were originally, and your changes should add new lines below them

File: scripts/convert-dlrs-rules.apex [hunk 2]
```diff
@@ -34,6 +30,37 @@ for (dlrs__LookupRollupSummary2__mdt dlrsRule : dlrs__LookupRollupSummary2__mdt.
...
     }
   }
+   if(invalidChildren.contains(dlrsRule.dlrs__ChildObject__c)){
+    //we will debug a list of unmigrateable rules to assist with the creation of the flow actions
+    Map<String,String> unmigratableRule = new Map<String,String>();
+    if(dlrsRule.dlrs__RelationshipCriteria__c != null){
+      unmigratableRule.put('SOQL Where Clause To Exclude Calc Items',dlrsRule.dlrs__RelationshipCriteria__c);
+    }
+    unmigratableRule.put('Records to rollup','Same as Prior records to rollup');
+    unmigratableRule.put('Prior records to rollup','provide the collection of child records (assuming rollup not from parent record else collection of parent records)');
```
Comment (jamessimone) [inline scripts/convert-dlrs-rules.apex:40]: This description for records to rollup and prior records to rollup is incorrect. Your description for `Prior records to rollup` belongs instead with `Records to rollup`. The description for `Prior records to rollup` should be something like: "A collection variable with `{!$Record__Prior}` in it, when using after update or after create and update flows".

File: scripts/convert-dlrs-rules.apex [hunk 3]
```diff
@@ -34,6 +30,37 @@ for (dlrs__LookupRollupSummary2__mdt dlrsRule : dlrs__LookupRollupSummary2__mdt.
...
+    }
+    unmigratableRule.put('Rollup target\'s SObject Name',dlrsRule.dlrs__ParentObject__c);    
+    unmigratableRule.put('Rollup Operation',operation.toUpperCase());
+    unmigratableRule.put('Rollup Object Field',dlrsRule.dlrs__AggregateResultField__c);   
+    unmigratableRule.put('Rollup Context','INSERT / UPDATE / UPSERT / DELETE: see documentation');
+    unmigratableRule.put('Lookup Field On Rollup Object','Id');
+    unmigratableRule.put('Lookup Field On Calc Item',dlrsRule.dlrs__RelationshipField__c);
+    unmigratableRule.put('Calc Item Rollup Field',dlrsRule.dlrs__FieldToAggregate__c);
+    unmigratableRule.put('Object for \"Prior records to rollup\"',dlrsRule.dlrs__ChildObject__c);
+    unmigratableRule.put('Action label',(String) dlrsRule.get('MasterLabel'));
```
Comment (jamessimone) [inline scripts/convert-dlrs-rules.apex:55]: see below comment - this line should become:

```java
unmigratableRule.put('Action label', customMetadata.label);
```

File: scripts/convert-dlrs-rules.apex [hunk 4]
```diff
@@ -34,6 +30,37 @@ for (dlrs__LookupRollupSummary2__mdt dlrsRule : dlrs__LookupRollupSummary2__mdt.
       operation = dlrsRule.dlrs__AggregateOperation__c;
     }
   }
+   if(invalidChildren.contains(dlrsRule.dlrs__ChildObject__c)){
+    //we will debug a list of unmigrateable rules to assist with the creation of the flow actions
```
Comment (jamessimone) [inline scripts/convert-dlrs-rules.apex:34]: comments should have a space following the `//` to aid in readability

**Review by jamessimone — CHANGES_REQUESTED** [review]
@mickster04 - I've added some comments that I'd like to see resolved prior to approving this pull request. Furthermore, I'd also like to see you successfully get prettier up and running. If you have npm and have installed the dependencies for this repository (which should be as simple as running `npm install` on the command line from within the repository's directory), you should be able to run this script to have the prettier formatting be applied to your changes by following these steps:

1. Add prettier to the `package.json` scripts section:
```json
  "scripts": {
    "test": "npm run testApex && npm run testLWC",
    "testApex": "sh ./scripts/runLocalTests.sh",
    "testLWC": "sfdx-lwc-jest",
    "prettier": "prettier"
  }
```

2. Invoke the npm script with the following command on your command line:

```
npm run prettier -prettier --write scripts/convert-dlrs-rules.apex
```

It's fine if you commit the changes to the `package.json` scripts section as well; I think that will be helpful for other people. Looking forward to seeing this PR fixed up so I can approve it - thanks for submitting!

**Comment by jamessimone** [comment]
@mickster04 once you’ve made those changes, please also update this PR to point to `v1.2.21` instead of to the main branch! Thank you!

File: README.md [hunk 5]
```diff
@@ -789,4 +793,4 @@ Further instructions for contributions are listed [in the Contributing doc](./Co
 
 ## Contact Info
 
-Drop me <a href="mailto:james@sheandjim.com" title="Email me">a line</a>, <a title="contact me" href="https://www.jamessimone.net/contact/">contact me online</a> or raise an issue here with questions.
\ No newline at end of file
+Drop me <a href="mailto:james@sheandjim.com" title="Email me">a line</a>, <a title="contact me" href="https://www.jamessimone.net/contact/">contact me online</a> or raise an issue here with questions.
```
[the remark this answers is not in the document]
Comment (mickster04) [reply README.md:796]: This came from my IDE formatting the readme, I didn't change this, but I also can't change the extra line appearing as it formats on save :/

File: scripts/convert-dlrs-rules.apex (hunk 1 above)
Comment (mickster04) [reply scripts/convert-dlrs-rules.apex:62]: I figured it was sensible to not bother starting the metadata variable if the rule wasn't going to be migrated?

File: scripts/convert-dlrs-rules.apex (hunk 2 above)
Comment (mickster04) [reply scripts/convert-dlrs-rules.apex]: thanks will fix

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (jamessimone) [reply README.md:798]: Ah interesting. Ok, I’ll take a look!

**Pull request merged**